lib/model/store.dart

```diff
@@ -1044,48 +1044,46 @@ class UpdateMachine {
           if (_disposed) return;
         } catch (e) {
           if (_disposed) return;
-
           store.isLoading = true;
-          bool isUnexpected;
+
+          if (e is! ApiRequestException) {
+            // Some unexpected error, outside even making the HTTP request.
+            // Definitely a bug in our code.
```

Comment:
Looks like this check also enables the exhaustive switch later. That's pretty neat!

Reply:
Yep!

Ideally I'd have liked to express this as one more case in the switch, so that the `rethrow` cases were all next to each other. But the patterns minilanguage doesn't seem to have a way to express that. (For example if it had negations of patterns, that would do it.) So this works instead.